src/services/blocks/BlocksService.ts

```diff
@@ -500,14 +536,29 @@ export class BlocksService extends AbstractService {
 				version.specVersion.toNumber(),
 			];
 
-			runtimeDoesNotMatch =
-				specName !== api.runtimeVersion.specName.toString() ||
-				specVersion !== api.runtimeVersion.specVersion.toNumber();
+			const hardcodedChainIncluded = hardcodedChains.includes(specName);
 
-			if (runtimeDoesNotMatch) {
+			if (!hardcodedChainIncluded) {
 				const metadata = await api.rpc.state.getMetadata(parentParentHash);
 
-				decorated = expandMetadata(api.registry, metadata);
+				/**
+				 * Check runtime associated with the cached decorated metadata
+				 * and compare it to the current specVersion
+				 */
+				if (that.cache.runtimeVersion === specVersion) {
+					decorated = that.cache.decorated;
+				} else {
+					// Decorate current metadata to read baseweight used in calcFee
+					decorated = expandMetadata(api.registry, metadata);
+
+					const cachedObject = this.extractWeightFromDecorated(
+						decorated,
+						specVersion
+					);
+
+					// Calls the cache setter
+					that.cache = { ...cachedObject };
```

Comment:
See comment above - I think this should be local to this service.

One idea is maybe just initialize the cache when the service starts. So in the config.options it could pass down the chains spec name and then based on the specName it grabs the correct `metaConstsCache` with `getBlockWeight`. We won't be able to inject the `metaConstsCache` the same way for testing, but it seems less convoluted